tests/spam_methods_test.py

```diff
@@ -23,6 +23,29 @@
     CorrectionMethod,
 )
 from qermit.taskgraph import CircuitShots  # type: ignore
+from qermit.spam import gen_UnCorrelated_SPAM_MitRes
+from qermit import CircuitShots
+from pytket import Circuit
+from pytket.extensions.quantinuum import QuantinuumBackend
+from .test_backends import MockQuantinuumBackend
+import pytest
+
+
+@pytest.mark.xfail(
+    reason=("Presently pytket removes unused qubits. There is also a missmatch between node and qubit names with QuantinuumBackend")
```

Comment:
There is no current plan to stop the removal of unused qubits in compilation, if that's what is causing this test to fail then it will still fail after changes made to the compilation.

Reply:
That pytket removes idle qubits is the cause of this failure on the develop branch. I think this is avoided by your repair here though -> https://github.com/CQCL/Qermit/pull/75. I imagined that this PR would go onto develop before that one though, so that we can check if these tests pass with the repairs.

Reply:
Though my understanding is that that problem has been worked around, and the remaining problem is due to the CompilationUnit only?
Please change the reason argument to "CompilationUnit in default compilation for pytket-quantinuum Backends does not correctly update its initial and final map objects"

Reply:
The `CompilationUnit` problem relates to the additional test added to `zne_test.py`. The remaining problem here relates the difference in the names of the nodes on the backend and the name off the qubits assigned by the default compilation pass of the `QuantinuumBackend`. I'm not sure which PR corrects that (it may be the same one you point to and I'm getting my wires crossed?), but I understood that a new release of `pytket-quantinuum` would fix this? In either case I believe that at the time of writing this message describes the problems, at least until there is a new release of `pytket-quantinuum`? Indeed this `xfail` should be removed in https://github.com/CQCL/Qermit/pull/75 as it should pass then, so the message is only really a temporary place holder.